### ISSUE TITLE
[Bug] `multilingual-e5-small` missing token_type_ids

### ISSUE THREAD

**Issue by BrianHillis-DG** [xenova/transformers.js#324]
**Describe the bug**
*A clear and concise description of what the bug is.*

When attempting to extract embeddings w/ `pipeline('feature-extraction', 'Xenova/multilingual-e5-small')`,the extractor produces `Error: An error occurred during model execution: "Missing the following inputs: token_type_ids.`

**How to reproduce**
*Steps or a minimal working example to reproduce the behavior*

``` javascript
import { pipeline } from '@xenova/transformers';
const extractor = await pipeline('feature-extraction', 'Xenova/multilingual-e5-small');
const embedding = await extractor('Test', {pooling: 'mean', normalize: true}); // Error produced here 
```
**Expected behavior**
*A clear and concise description of what you expected to happen.*

I expect the pipeline to run without producing errors

**Logs/screenshots**
*If applicable, add logs/screenshots to help explain your problem.*

```
Error: An error occurred during model execution: "Missing the following inputs: token_type_ids.
    at validateInputs (webpack-internal:///(rsc)/./node_modules/@xenova/transformers/src/models.js:320:15)
    at sessionRun (webpack-internal:///(rsc)/./node_modules/@xenova/transformers/src/models.js:343:33)
    at Function.encoderForward [as _forward] (webpack-internal:///(rsc)/./node_modules/@xenova/transformers/src/models.js:566:18)
    at Function.forward (webpack-internal:///(rsc)/./node_modules/@xenova/transformers/src/models.js:806:27)
    at Function._call (webpack-internal:///(rsc)/./node_modules/@xenova/transformers/src/models.js:797:27)
    at Function.closure [as model] (webpack-internal:///(rsc)/./node_modules/@xenova/transformers/src/utils/core.js:68:28)
    at Function._call (webpack-internal:///(rsc)/./node_modules/@xenova/transformers/src/pipelines.js:104:34)
    at Function._call (webpack-internal:///(rsc)/./node_modules/@xenova/transformers/src/pipelines.js:810:45)
    at closure (webpack-internal:///(rsc)/./node_modules/@xenova/transformers/src/utils/core.js:68:28)
```

**Environment**
- Transformers.js version: `"@xenova/transformers": "^2.6.1",`
- Browser (if applicable): N/A
- Operating system (if applicable): N/A
- Other: 

**Additional context**
*Add any other context about the problem here.*

Not seeing the error with `Xenova/beta-base-uncased` or `Xenova/all-MiniLM-L6-v2` with exactly the same setup.

**Comment by xenova** [comment]
Hi there 👋  This has already been reported [here](https://github.com/xenova/transformers.js/issues/267).

See here for the fix: https://github.com/xenova/transformers.js/issues/267#issuecomment-1697858582
TLDR: The model author decided to mix different models and tokenizers.

config.json:
```json
{
  "architectures": [
    "BertModel"
  ],
  "tokenizer_class": "XLMRobertaTokenizer",
}
```

workaround:
```js
const extractor = await pipeline(
    "feature-extraction",
    "Xenova/multilingual-e5-small",
    { revision: "main" }
);

extractor.tokenizer.prepare_model_inputs = (model_inputs) => {
    const token_type_ids = model_inputs.attention_mask.clone();
    token_type_ids.data.fill(0n);
    return {...model_inputs, token_type_ids};
};
const result = await extractor("This is a simple test.", {
    pooling: "mean",
    normalize: true,
});
console.log(result);
// Tensor {
//   data: Float32Array(384) [0.019251275807619095, 0.008903810754418373, -0.05832963064312935, ...]
//   dims: [1, 384],
//   size: 384,
//   type: "float32"
// }
```

**Comment by BrianHillis-DG** [comment]
Apologies for the duplicate report! This solution works great for me, many thanks! :bow:
